webhook/resources/recurringjob/mutator.go

```diff
@@ -52,6 +52,9 @@ func (r *recurringJobMutator) Create(request *admission.Request, newObj runtime.
 	if name != recurringjob.Name {
 		patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/metadata/name", "value": "%s"}`, name))
 	}
+	if recurringjob.Spec.Name == "" {
+		patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/name", "value": "%s"}`, recurringjob.Name))
+	}
```

Comment:
Why do we need the spec.Name?

Reply:
IIRC it's a placeholder for the full name before auto-correction.